docs/dev/relay_op_strategy.rst

```diff
@@ -0,0 +1,256 @@
+
+.. code:: c
+
+    OpStrategy(const Attrs& attrs, const Array<Tensor>& inputs, const Type& out_type, const Target& target)
+
+, that the function returns an ``OpStrategy`` given the op attributes, input
+tensors, output types, and target to compile to,
+
+
+
+Register strategy for a new operator
+------------------------------------
+
+There are three methods to register a strategy function for an operator,
+defined in ``python/tvm/relay/op/op.py``.
+
+First, for operators that have injective, broadcast, or reduction pattern, we
+can call ``register_injective_schedule``, ``register_broadcast_schedule``, and
+``register_reduce_schedule`` repsectively. The schedule function for these
+patterns are already registered by each target and can be applied to these
+operators. We assume the compute function should be same across all targets, and
+``FTVMCompute`` needs to be registered to the op before invoking register
+schedule.
+
+.. code:: python
+
+    register_injective_schedule("my_new_op")
+
+Second, for operators that doesn't have these common patterns mentioned before,
+but also have the same compute function for all targets, we can use
+``register_schedule`` API. But before that, we need to first define the
+``FTVMSchedule`` function as follows:
+
+.. code:: python
+
+    # add to python/tvm/relay/op/strategy/generic.py
+    @generic_func
+    def schedule_my_new_op(attrs, outs, target):
+        ...
+
+    # add to each target file in python/tvm/relay/op/strategy, e.g., x86.py, cuda.py, etc.
+    @schedule_my_new_op.register("cpu")
+    def schedule_my_new_op_cpu(attrs, outs, target):
+        ...
+
+Now that we've created the ``FTVMSchedule`` for this new operator, we can
+register the strategy using ``register_schedule``:
+
+.. code:: python
+
+    register_schedule("my_new_op", strategy.schedule_my_new_op)
+
+Third, for most comprehensive usage of op strategy, we can allow operator to use
+different implementation for both compute and schedule for different targets.
+Similarly, we need to first define the ``FTVMStrategy`` function as follows:
+
+.. code:: python
+
+    # add to python/tvm/relay/op/strategy/generic.py
+    @override_native_generic_func("my_new_op_strategy")
+    def my_new_op_strategy(attrs, inputs, out_type, target):
+        strategy = _op.OpStrategy()
+        strategy.add_implementation(
+            wrap_compute_my_new_op(topi.my_new_op),
+            wrap_topi_schedule(topi.generic.schedule_my_new_op),
+            name="my_new_op.generic")
+        return strategy
+
+    # add to each target file in python/tvm/relay/op/strategy, e.g., x86.py, cuda.py, etc.
+    @my_new_op_strategy.register("cpu")
+    def my_new_op_strategy_cpu(attrs, inputs, out_type, target):
+        strategy = _op.OpStrategy()
+        strategy.add_implementation(
+            wrap_compute_my_new_op(topi.x86.my_new_op),
+            wrap_topi_schedule(topi.x86.schedule_my_new_op),
+            name="my_new_op.generic")
+        return strategy
+
```

Comment:
Maybe we can move the introduction to `add_implementation` here?

Reply:
moved the introduction of API before this block.